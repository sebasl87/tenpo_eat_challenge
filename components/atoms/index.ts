export { default as RestaurantsLayout } from './RestaurantsLayout';
