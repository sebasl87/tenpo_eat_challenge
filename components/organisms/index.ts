export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
