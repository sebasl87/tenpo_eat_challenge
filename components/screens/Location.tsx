import React, { useEffect, useState } from 'react'
import { Platform, StyleSheet, View } from 'react-native';
import MapView, { Marker, PROVIDER_GOOGLE } from 'react-native-maps';
import { check, request, PERMISSIONS, RESULTS } from "react-native-permissions" // ðŸ‘ˆ
import Geolocation from "react-native-geolocation-service"

export interface IGoogleMaps {
    latitude: number;
    longitude: number;
}

function Location() {
    const [location, setLocation] = useState<IGoogleMaps>()

    const handleLocationPermission = async () => {
        let permissionCheck = '';
        if (Platform.OS === 'ios') {
            permissionCheck = await check(PERMISSIONS.IOS.LOCATION_WHEN_IN_USE);

            if (
                permissionCheck === RESULTS.BLOCKED ||
                permissionCheck === RESULTS.DENIED
            ) {
                const permissionRequest = await request(
                    PERMISSIONS.IOS.LOCATION_WHEN_IN_USE,
                );
                permissionRequest === RESULTS.GRANTED
                    ? console.warn('Location permission granted.')
                    : console.warn('location permission denied.');
            }
        }

        if (Platform.OS === 'android') {
            permissionCheck = await check(PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION);

            if (
                permissionCheck === RESULTS.BLOCKED ||
                permissionCheck === RESULTS.DENIED
            ) {
                const permissionRequest = await request(
                    PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION,
                );
                permissionRequest === RESULTS.GRANTED
                    ? console.warn('Location permission granted.')
                    : console.warn('location permission denied.');
            }
        }
    };

    useEffect(() => {
        handleLocationPermission()
    }, [])

    useEffect(() => {
        Geolocation.getCurrentPosition(
            position => {
                const { latitude, longitude } = position.coords
                setLocation({ latitude, longitude })
            },
            error => {
                console.log(error.code, error.message)
            },
            { enableHighAccuracy: true, timeout: 15000, maximumAge: 10000 }
        )
    }, [])

    return (
        <View
            style={styles.mapStyle}
        >
            {location && (<MapView
                style={styles.mapStyle}
                provider={PROVIDER_GOOGLE}
                initialRegion={{
                    latitude: location.latitude,
                    longitude: location.longitude,
                    latitudeDelta: 0.0922,
                    longitudeDelta: 0.0421,
                }}
                showsUserLocation={true}
            >
                <Marker
                    image={require("../../assets/icons/pointer.png")}
                    style={{ height: 50 }}
                    coordinate={{
                        latitude: location.latitude,
                        longitude: location.longitude,
                    }}
                    title={'Test Marker'}
                />
            </MapView>)}
        </View>
    );
}
const styles = StyleSheet.create({

    container: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
        alignItems: 'center',
        justifyContent: 'flex-end',
    },
    mapStyle: {
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        bottom: 0,
    },
});
export default Location