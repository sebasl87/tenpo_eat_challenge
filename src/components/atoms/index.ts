import AddressLayout from './AddressLayout';
export { default as RestaurantsLayout } from './RestaurantsLayout';
export { default as CategoriesLayout } from './CategoriesLayout';
export { default as HeaderFavorites } from './HeaderFavorites';
export { default as AddressLayout } from './AddressLayout';
