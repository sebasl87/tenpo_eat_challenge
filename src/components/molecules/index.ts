export { default as DetailCard } from './DetailCard';
export { default as DiscountCircle } from './DiscountCircle';
export { default as LogoRounded } from './LogoRounded';
export { default as DetailFavorites } from './DetailFavorites';
