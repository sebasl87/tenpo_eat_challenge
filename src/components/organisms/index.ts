export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
