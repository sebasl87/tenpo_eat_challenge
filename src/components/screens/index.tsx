export { default as HomeScreen } from "./HomeScreen";
export { default as Location } from "./Location";
export { default as RestaurantDetail } from "./RestaurantDetail";