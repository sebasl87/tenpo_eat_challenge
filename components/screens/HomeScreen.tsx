import React from 'react'
import { Restaurant } from '../organisms'


function HomeScreen() {
    return (
        <>
            <Restaurant />
        </>
    );
}

export default HomeScreen;