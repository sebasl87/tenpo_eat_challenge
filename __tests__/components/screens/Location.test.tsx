import React from "react"
import renderer from "react-test-renderer";

import { Location } from "../../../src/components/screens"

describe("<Location />", () => {

    it("should render successfully", () => {
        const restaurantSection = renderer
            .create(<Location />)
            .toJSON();

        expect(restaurantSection).toMatchSnapshot();
    });

    it("should render successfully with location", () => {
        jest.mock("react-redux", () => ({
            ...jest.requireActual("react-redux"),
            useDispatch: jest.fn(),
            useSelector: jest.fn()
                .mockReturnValueOnce({
                    address: {
                        address: { formated_address: "Some mock street, Santiago, Chile" },
                        geolocation: { lat: -70, lng: -34 }
                    }
                }),
        }));
        const restaurantSection = renderer
            .create(<Location />)
            .toJSON();

        expect(restaurantSection).toMatchSnapshot();
    });
})