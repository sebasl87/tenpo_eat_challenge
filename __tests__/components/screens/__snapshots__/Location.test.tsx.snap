// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Location /> should render successfully 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ADADAD",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
          "opacity": 0.5,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/imgs/wait.gif",
          }
        }
        style={
          {
            "height": 150,
            "width": 300,
          }
        }
      />
    </View>
  </Modal>,
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": "25%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#ADADAD",
          "fontFamily": "Roboto-light",
          "fontSize": 18,
          "paddingTop": 3,
        }
      }
    >
      Esperando tu ubicación...
    </Text>
  </View>,
]
`;

exports[`<Location /> should render successfully with location 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ADADAD",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
          "opacity": 0.5,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/imgs/wait.gif",
          }
        }
        style={
          {
            "height": 150,
            "width": 300,
          }
        }
      />
    </View>
  </Modal>,
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": "25%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#ADADAD",
          "fontFamily": "Roboto-light",
          "fontSize": 18,
          "paddingTop": 3,
        }
      }
    >
      Esperando tu ubicación...
    </Text>
  </View>,
]
`;
